### PR TITLE
(chore) Bump carbon version and clean up some redundant components

### PR DESCRIPTION
**Pull request by CynthiaKamau**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes the ticket number in the format `OHRI-123 My PR title`.
- [ ] My work includes tests or is validated by existing tests.

## Summary

This pr addresses the following :
- Bump carbon version
- Bumping the openmrs version
-  Clean up some redundant components that now exist in carbon
- Removed unused imports and routes after homepage migration
- Update the jest config to ignore and be aware of important files
- Add new eslint rules for files

### CODE REVIEW

**Review by hadijahkyampeire — COMMENTED** [review]
This looks good @CynthiaKamau , I was expecting to see the carbon version changed in the package.json, doesn't bumping change it?

File: package.json [hunk 1]
```diff
@@ -102,7 +102,6 @@
     "webpack-dev-server": "^4.8.1"
   },
   "resolutions": {
-    "@carbon/react": "1.17.0",
```
Comment (arodidev) [inline package.json:105]: I may not have full context on what's happening here, why was this removed?

Comment (CynthiaKamau) [reply package.json:105]: Its an older version and is used by default

**Review by arodidev — APPROVED** [review]
Thanks @CynthiaKamau
